File: Shared/Models/TrainLine.swift
import SwiftUI

// Línea de tren. id == gerenciaId de la API SOFSE.
struct TrainLine: Identifiable, Hashable {
    let id: Int
    let nombre: String
    let shortCode: String
    let colorHex: String
    let covered: Bool

    var color: Color { Color(hex: colorHex) }
}

extension TrainLine {
    // Las 9 líneas de lineas.json + paleta del PLAN.
    static let all: [TrainLine] = [
        TrainLine(id: 1,   nombre: "Sarmiento",        shortCode: "SA", colorHex: "#B83280", covered: true),
        TrainLine(id: 5,   nombre: "Mitre",            shortCode: "MI", colorHex: "#1E7FD4", covered: true),
        TrainLine(id: 11,  nombre: "Roca",             shortCode: "RO", colorHex: "#16A34A", covered: true),
        TrainLine(id: 21,  nombre: "Belgrano Sur",     shortCode: "BS", colorHex: "#EAB308", covered: true),
        TrainLine(id: 31,  nombre: "San Martín",       shortCode: "SM", colorHex: "#E0632B", covered: true),
        TrainLine(id: 41,  nombre: "Tren de la Costa", shortCode: "TC", colorHex: "#0D9488", covered: true),
        TrainLine(id: 61,  nombre: "Belgrano Norte",   shortCode: "BN", colorHex: "#8A94A6", covered: false),
        TrainLine(id: 71,  nombre: "Urquiza",          shortCode: "UR", colorHex: "#8A94A6", covered: false),
        TrainLine(id: 501, nombre: "Regionales",       shortCode: "RG", colorHex: "#7C5CFC", covered: true),
    ]

    static let unknown = TrainLine(id: -1, nombre: "Sin línea", shortCode: "?", colorHex: "#8A94A6", covered: false)

    static func line(id: Int) -> TrainLine {
        all.first { $0.id == id } ?? .unknown
    }

    static func line(nombre: String?) -> TrainLine {
        guard let nombre else { return .unknown }
        let q = nombre.folding(options: [.diacriticInsensitive, .caseInsensitive], locale: Locale(identifier: "es"))
        return all.first {
            $0.nombre.folding(options: [.diacriticInsensitive, .caseInsensitive], locale: Locale(identifier: "es")) == q
        } ?? .unknown
    }
}

File: android/app/src/main/java/com/alandaitch/anden/data/model/TrainLine.kt
package com.alandaitch.anden.data.model

import androidx.compose.ui.graphics.Color
import com.alandaitch.anden.ui.theme.hexColor
import com.alandaitch.anden.util.Text

// Línea de tren. id == gerenciaId de la API SOFSE.
data class TrainLine(
    val id: Int,
    val nombre: String,
    val shortCode: String,
    val colorHex: String,
    val covered: Boolean
) {
    val color: Color get() = hexColor(colorHex)

    companion object {
        // Las 9 líneas de lineas.json + paleta del PLAN.
        val all: List<TrainLine> = listOf(
            TrainLine(1, "Sarmiento", "SA", "#B83280", true),
            TrainLine(5, "Mitre", "MI", "#1E7FD4", true),
            TrainLine(11, "Roca", "RO", "#16A34A", true),
            TrainLine(21, "Belgrano Sur", "BS", "#EAB308", true),
            TrainLine(31, "San Martín", "SM", "#E0632B", true),
            TrainLine(41, "Tren de la Costa", "TC", "#0D9488", true),
            TrainLine(61, "Belgrano Norte", "BN", "#8A94A6", false),
            TrainLine(71, "Urquiza", "UR", "#8A94A6", false),
            TrainLine(501, "Regionales", "RG", "#7C5CFC", true),
        )

        val unknown = TrainLine(-1, "Sin línea", "?", "#8A94A6", false)

        fun line(id: Int): TrainLine = all.firstOrNull { it.id == id } ?: unknown

        fun line(nombre: String?): TrainLine {
            if (nombre == null) return unknown
            val q = Text.normalize(nombre)
            return all.firstOrNull { Text.normalize(it.nombre) == q } ?: unknown
        }
    }
}
